v2/protocol/pubsub/message.go

```diff
@@ -0,0 +1,123 @@
+package pubsub
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"strings"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/binding/format"
+	"github.com/cloudevents/sdk-go/v2/binding/spec"
+)
+
+const (
+	prefix      = "ce-"
+	contentType = "Content-Type"
+)
+
+var specs = spec.WithPrefix(prefix)
+
+// Message represents a Pub/Sub message.
+// This message *can* be read several times safely
+type Message struct {
+	Msg pubsub.Message
+
+	format   format.Format
+	version  spec.Version
+	encoding binding.Encoding
```

Comment:
You can have `format` and `version` mutually exclusive to identify the encoding, without saving it as field of the struct: https://github.com/cloudevents/sdk-go/blob/master/v2/protocol/kafka_sarama/message.go#L78

Reply:
done